client/blocks/term-form-dialog/index.jsx

```diff
@@ -175,12 +174,10 @@ class TermFormDialog extends Component {
 		} );
 	}
 
-	UNSAFE_componentWillReceiveProps( newProps ) {
-		if (
-			this.props.term !== newProps.term ||
-			( this.props.showDialog !== newProps.showDialog && newProps.showDialog )
-		) {
-			this.init( newProps );
+	componentDidUpdate( prevProps ) {
+		const { term, showDialog } = this.props;
+		if ( term !== prevProps.term || ( showDialog !== prevProps.showDialog && showDialog ) ) {
+			this.init( this.props );
```

Comment:
With this change, all `init()` calls now work with `this.props`. This means we can essentially remove this argument and retrieve `this.props` directly inside `init()`.

Reply:
Yup, addressed in https://github.com/Automattic/wp-calypso/pull/57822/commits/eaca08848134ef50b3077daedf6784320a40e032